File: abc/abc234/a.cpp
// https://atcoder.jp/contests/abc234/submissions/28401817

#include <bits/stdc++.h>
using namespace std;
int main() {
  int t;
  cin >> t;
  long long f1 = t * t + 2 * t + 3;
  long long f2 = f1 + t;
  long long f3 = (f2) * (f2) + 2 * (f2) + 3;
  long long f4 = f1 * f1 + 2 * f1 + 3;
  long long f5 = (f3 + f4) * (f3 + f4) + 2 * (f3 + f4) + 3;
  cout << f5;
}

File: abc/abc234/a.cs
// https://atcoder.jp/contests/abc234/submissions/46503255

using System;

public class A
{
  static int f(int x)
  {
    return x * x + 2 * x + 3;
  }

  static void Main()
  {
    int x = int.Parse(Console.ReadLine());
    Console.WriteLine(A.f(A.f(A.f(x) + x) + A.f(A.f(x))));
  }
}

